scripts/extractions/extract_sar.py

```diff
@@ -0,0 +1,400 @@
+"""Extract S1 data using OpenEO-GFMAP package."""
+import argparse
+import json
+import logging
+import os
+from datetime import datetime
+from functools import partial
+from importlib.metadata import version
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from typing import List, Optional
+
+import geojson
+import geopandas as gpd
+import openeo
+import pandas as pd
+import pystac
+import requests
+import xarray as xr
+from openeo_gfmap import Backend, BackendContext, FetchType, TemporalContext
+from openeo_gfmap.backend import cdse_connection
+from openeo_gfmap.fetching.s1 import build_sentinel1_grd_extractor
+from openeo_gfmap.manager import _log
+from openeo_gfmap.manager.job_manager import GFMAPJobManager
+from openeo_gfmap.manager.job_splitters import (
+    _append_h3_index,
+    _load_s2_grid,
+    split_job_s2grid,
+)
+from openeo_gfmap.stac import AUXILIARY
+from shapely.geometry import Point
+
+# Logger for this current pipeline
+_pipeline_log: Optional[logging.Logger] = None
+
+
+def _setup_logger(level=logging.INFO) -> None:
+    global _pipeline_log
+    """Setup the logger from the openeo_gfmap package to the assigned level."""
+    _pipeline_log = logging.getLogger("pipeline_sar")
+
+    _pipeline_log.setLevel(level)
+    _log.setLevel(level)
+
+    stream_handler = logging.StreamHandler()
+    _log.addHandler(stream_handler)
+    _pipeline_log.addHandler(stream_handler)
+
+    formatter = logging.Formatter("%(asctime)s|%(name)s|%(levelname)s:  %(message)s")
+    stream_handler.setFormatter(formatter)
+
+    # Exclude the other loggers from other libraries
+    class ManagerLoggerFilter(logging.Filter):
+        """Filter to only accept the OpenEO-GFMAP manager logs."""
+
+        def filter(self, record):
+            return record.name in [_log.name, _pipeline_log.name]
+
+    stream_handler.addFilter(ManagerLoggerFilter())
+
+
+def _buffer_geometry(
```

Comment:
why is this not part of GFMAP?

Reply:
`_buffer_geometry` here is related to the use-case. We can have an implementation in GFMAP but I would wait for the implementation of the pipeline from the data engineering team